### PR TITLE
Add bad file extension error message

### PR DESCRIPTION
**Pull request by ghost**
ntdkhiem and my attempt at blocking the user from opening forbidden file extensions. Currently sends an error message popup that sends user back to the text editing area; would be better to remain in scroll menu.

### CODE REVIEW

**Comment by ntdkhiem** [comment]
We should finish up the remaining tasks before actually merge this.

File: src/text/text_editor.py [hunk 1]
```diff
@@ -342,15 +344,21 @@ async def coroutine() -> None:
...
     app = get_app()
 
+    # Put current window in a temp variable
     focused_before = app.layout.current_window
+    # Focus cursor to the given dialog
     app.layout.focus(dialog)
+    # Wait for the dialog to finish (returns None)
     result = await dialog.future
+    # Re-focus cursor back to window in temp variable
     app.layout.focus(focused_before)
```
Comment (BoraxTheClean) [inline src/text/text_editor.py:359]: Hell yes, nice comments!

File: tests/basic_tests.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+@pytest.fixture(autouse=True, scope="function")
+def mock_input() -> None:
+    """Fixture class for user input"""
+    pipe_input = create_pipe_input()
+    try:
+        with create_app_session(input=pipe_input, output=DummyOutput()):
+            yield pipe_input
+    finally:
+        pipe_input.close()
```
Comment (BoraxTheClean) [inline tests/basic_tests.py]: What does this do? Does it just facilitate inputing text with other tests? Maybe add some comments.

File: src/scroll/scroll_menu.py [hunk 3]
```diff
@@ -39,15 +39,22 @@ def set_cancel() -> None:
...
+                    with open(self.cur_file_path, "r") as f:
+                        f_content = f.read()
+                    text_editor.set_text_field(f_content)
+                else:
+                    text_editor.show_message(
+                        title="extension_error",
+                        text="Unsupported file extension. Only '.txt' and '.md' are supported",
+                    )
 
+            self.future.set_result(None)
```
Comment (BoraxTheClean) [inline src/scroll/scroll_menu.py]: Commenting this method and improving the docstring would be pretty dece.

**Comment by BoraxTheClean** [comment]
> We should finish up the remaining tasks before actually merge this.

Which Issues is this blocked by?

Long lived feature branches are kind of an anti-pattern. So we should be prioritizing merging over waiting whenever possible.

**Comment by ntdkhiem** [comment]
@BoraxTheClean , we're done. Could you take a look?

File: .gitignore [hunk 4]
```diff
@@ -31,3 +31,5 @@ build/
 .DS_Store
 # Vim
 .vim/
+# ThoughtBox
```
Comment (BoraxTheClean) [inline .gitignore]:
```suggestion
# ThoughtBox's Internal Application File System
```

File: src/scroll/scroll_menu.py [hunk 5]
```diff
@@ -39,15 +39,25 @@ def set_cancel() -> None:
             """Cancel don't open file"""
             self.future.set_result(None)
 
+        # Raise exception if attempt to open any extension besides .txt or .md
         def set_done() -> None:
             """Future object when done return None"""
```
Comment (BoraxTheClean) [inline src/scroll/scroll_menu.py]:
```suggestion
            """Handles actions related to loading previews of files into the file explorer."""
```
Is this an accurate docstring for this function?

**Review by BoraxTheClean — APPROVED** [review]
This looks good. I think we should make a ticket for the backlog to clean up the code in:

`def set_done() -> None:` and similar functions. They seem to work nicely, but they are hacky and complex to reason about.

File: src/scroll/scroll_menu.py (hunk 5 above)
Comment (ntdkhiem) [reply src/scroll/scroll_menu.py]: I think it should be `handles actions related to adding file's contents to text editor`

Comment (BoraxTheClean) [reply src/scroll/scroll_menu.py]: Oh, the code in `scroll_menu.py` is loading files into the text editor? That seems kind of backwards. Maybe add an issue to refactor? Definitely add that docstring too.

**Comment by ghost** [comment]
We completed the three features from the original issue:

1. Open the app in a directory called .`thought_box` if it exists, creating it if not.
2. Disable the .. button in the scrollbar menu in `.thought_box` itself to restrict the app to that directory and its children.
3. Prevent opening files with extensions other than `.txt` or `.md` and generate an error message if the user attempts it.

I noted two issues to fix:
1. The error message for loading a disallowed extension moves focus to the editing area from the scroll menu, instead of staying there.
2. The .. button doesn't appear for empty directories.

**Pull request merged**